### DISCUSSION TITLE
Upgrade of behavior on a live cluster

### DISCUSSION THREAD

**Discussion by zaibon** [Tochemey/goakt#379] [Q&A] [answered]
Hi 👋 

I'm evaluation Goakt for one of my project and I have some questions regarding the best practices.
Is there a define pattern to follow when you need to upgrade the behavior of some actor on a live cluster?

My idea to accomplish this is:
1. inject new node in the cluster that run a binary containing the code of the new behavior, 
2. Issue a command to upgrade the behavior or the actors to use the new code.
And all of this without dropping messages?
Is that the right approach or is there a define way to do something like that?

**Comment by Tochemey** [comment]
Hello @zaibon. Thanks for your question. I am afraid there is no define pattern to accomplish what you are suggesting. Also give me some time to understand what you want to accomplish.

**Comment by Tochemey** [comment]
@zaibon FYI actor behavior are kind of various way of handling messages and those behavior need to be known ahead of time and baked into the running binaries of the given actor. It is recommended to use behaviors pattern in conjunction with stashing. I hope this help.

**Reply by Tochemey** [reply]
@zaibon is there any particular use case that is common which such a pattern?. I will be glad if you can elaborate a bit more on it. Thank you.

**Reply by zaibon** [reply, marked as answer]
I'm just trying to figure out a way to be able to upgrade my code without downtime. 

Since we have the possibility to  spawn and stop actors at runtime, I figured I should be able to upgrade code by spawning new actor and stopping hold once. Probably by using a combination of actor name and version.
I've not tested any of this yet, it's only theoretical from what I could find in the documentation.

I wanted to get your view on it, see if this kind of pattern already existed.

**Reply by Tochemey** [reply]
@zaibon I understand. Always happy to help.
